qiskit_nature/drivers/second_quantization/electronic_structure_molecule_driver.py

```diff
@@ -81,7 +81,8 @@ def driver_class_from_type(
                 raise MissingOptionalLibraryError(
                     libname=driver_type, name="ElectronicStructureDriverType"
                 )
```

Comment:
Related to the logic above which is dealing with driver importing. Can this still occur? I only noticed because I assumed that during the move to lazy imports we can get rid of the Nature-specific `MissingOptionalLibraryError` entirely..

Reply:
Lazy imports is not about removing `MissingOptionalLibraryError` . It is just a delayed mechanism but stiil raises `MissingOptionalLibraryError` as was done before. That exception needs to be handled by other areas of code and it is raised here because the Molecule Driver has a search driver mechanism and will raise the same exception if no appropriate driver was found.

Reply:
So the lazy import mechanism does not provide a utility which we could use for this use case? In that case I'm obviously fine with this :+1:

Reply:
Yes, the Molecule Driver search for an appropriate driver based on an enumeration passed is our own custom implementation. There is nothing there for it but we raise the correct exception.